### PR TITLE
Adding new error and strict imitation compatibility loss functions for KL Divergence and Binary Cross Entropy losses

### PR DESCRIPTION
**Pull request by ilmarinen**
This PR adds the new error and strict imitation versions of the KL Divergence and Binary Cross Entropy loss functions.

### CODE REVIEW

File: backwardcompatibilityml/loss/strict_imitation.py [hunk 1]
```diff
@@ -133,7 +134,145 @@ def forward(self, x, y, reduction="mean"):
...
+
+    def __init__(self, h1, h2, lambda_c, discriminant_pivot=0.5, **kwargs):
+        super(StrictImitationBinaryCrossEntropyLoss, self).__init__()
+        self.h1 = h1
+        self.h2 = h2
+        self.loss = F.binary_cross_entropy
+        self.lambda_c = lambda_c
+        self.discriminant_pivot = discriminant_pivot
+
+    def dissonance(self, h1_output_sigmoid, h2_output_sigmoid):
```
Comment (ilmarinen) [inline backwardcompatibilityml/loss/strict_imitation.py:183]: @nushib If you take a look at how the dissonance is being calculated for new error vs strict imitation. The end result is the same, i.e. we are calculating how h2 differs from the target labeling in either case.

This makes me think that I have interpreted the strict imitation loss incorrectly, at least in the case of Binary Cross Entropy.

Should I instead just be measuring the negative_log_likelihood(h2, h1) in the cases where h1 is correct? For each and every strict imitation dissonance?

Or should i be measuring parent_loss(h2, h1) in cases where h1 is correct.

If the latter, then depending on the type of parent_loss, i.e. NLL, CrossEntropy, KLDiv or BinaryCrossEntropy ... how we define this is open to interpretation.

Comment (ilmarinen) [reply backwardcompatibilityml/loss/strict_imitation.py:183]: Based on discussion offline. Will have the dissonance in the Strict Imitation Binary Cross Entropy loss be calculated using the negative log likelihood. And will make sure to document why we are doing this so that users are aware.

**Review by nushib — APPROVED** [review]
